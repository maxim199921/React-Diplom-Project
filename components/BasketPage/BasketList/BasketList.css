.BasketList {
    margin-bottom: 20px;
    box-shadow: 0 5px 5px 0 rgba(22,35,23,.4);
    min-height: 140px;
    /*border-top: 1px solid rgba(22,35,23,.1);*/
}

.BasketList-img-wrapper {
    margin-bottom: 10px;
    margin-top: 10px;
    height: 120px;
    display: flex;
}

.BasketList-img {
    margin: auto;
    max-height: 100%;
    max-width: 100%;
}

.BasketList-info {
    text-align: center;
    position: relative;
}

.BasketList-count {
    position: absolute;
    right: 15px;
    top: 35%;
}

.BasketList-price {
    color: orange;
    font-size: 25px;
    font-weight: bold;
}

.BasketList-model {
    font-size: 22px;
    line-height: 26px;
    font-weight: bold;
    color: #130e27;
}

.BasketList-model {}

.BasketList-description{
    margin-top: 15px;
}

.BasketList-count-inp {
    background-color: #5dbbc1;
    color: white;
    width: 20px;
    border-radius: 10%;
    border: none;
    box-shadow: 0 0 5px 0 rgba(22,35,23,.7);
    padding: 0;
    margin: 0;
    line-height: 0;
    font-weight: bold;
    user-select: none;
}

.BasketList-count-minus {
    color: red;
    font-size: 25px;
    line-height: 0;
    font-weight: bold;
    margin-right: 5px;
    cursor: pointer;
    user-select: none;
}


.BasketList-count-plus {
    color: green;
    font-size: 25px;
    line-height: 0;
    font-weight: bold;
    margin-left: 2px;
    cursor: pointer;
    user-select: none;
}

@media (max-width: 1199px) {
    .BasketList .btn-info {
        font-size: 12px;
    }
    .BasketList-price {
        font-size: 18px;
    }
}

@media (max-width: 992px) {
    .BasketList .btn-info {
        font-size: 12px;
        margin-top: 20px;
        margin-bottom: 20px;
    }
    .BasketList-price {
        font-size: 18px;
        text-align: right;
    }
}