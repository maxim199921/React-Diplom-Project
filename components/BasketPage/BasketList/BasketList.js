import React from 'react';
import PropTypes from 'prop-types';

import './BasketList.css';
import {connect} from 'react-redux';
import {activeStatusSwitch, cntChange, cntStartValue} from '../../../redux/ActionCreaters/CatalogActions'

class BasketList extends React.PureComponent {

    static propTypes = {
        info: PropTypes.shape({
            id: PropTypes.number.isRequired,
            type: PropTypes.string.isRequired,
            img: PropTypes.string.isRequired,
            model: PropTypes.string.isRequired,
            price: PropTypes.string.isRequired,
            description: PropTypes.string.isRequired,
            status: PropTypes.bool.isRequired,
            cnt: PropTypes.number.isRequired,
        }),
    };

    cntPlus = () => {
        this.props.dispatch( cntChange(this.props.info.id, +1) );
    };

    cntMinus = () => {
        if (this.props.info.cnt >= 2) {
            this.props.dispatch( cntChange(this.props.info.id, -1) );
        }
    };

    deleteItemFromBasket = () => {
        let changeStatusStartList = JSON.parse(localStorage.startPriceList);
        changeStatusStartList[this.props.info.id -1].status = false;
        localStorage['startPriceList'] = JSON.stringify(changeStatusStartList);
        this.props.dispatch( activeStatusSwitch(this.props.info.id, false) );
        this.props.dispatch( cntStartValue(this.props.info.id, 1) );
    };


    render() {

        return (
            <div className="col-12 BasketList">
                <div className="row">
                    <div className="col-12 col-lg-2 BasketList-img-wrapper">
                        <img className="BasketList-img" src={this.props.info.img}/>
                    </div>
                    <div className="col-12 col-lg-8 BasketList-info align-self-center">
                        <div className="row">
                            <div className="col-lg-12 BasketList-model">{this.props.info.model}</div>
                            <div className="col-lg-12 BasketList-description">{this.props.info.description}</div>
                            <div className="BasketList-count">
                                <span className="BasketList-count-minus" onClick={this.cntMinus}>-</span>
                                <input type="text" className="BasketList-count-inp" value={this.props.info.cnt} disabled={true}/>
                                <span className="BasketList-count-plus" onClick={this.cntPlus}>+</span>
                            </div>
                        </div>
                    </div>
                    <div className="col-6 col-lg-1 BasketList-price align-self-center">{parseInt(this.props.info.price)*this.props.info.cnt}$</div>
                    <div className="col-6 col-lg-1 align-self-center">
                        <input type="button" className="btn btn-info btn-primary-add" value="delete"
                             onClick={this.deleteItemFromBasket}/>
                    </div>
                </div>
            </div>
        );

    }

}

const mapStateToProps = function (state) {
    return {};
};


export default connect(mapStateToProps)(BasketList);