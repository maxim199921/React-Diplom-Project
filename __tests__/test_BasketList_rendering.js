import React from 'react';
import renderer from 'react-test-renderer';
import BasketList from '../components/BasketPage/BasketList/BasketList';
import {HashRouter, hashHistory} from 'react-router-dom';
import {Provider} from 'react-redux';
import configureStore from '../redux/configureStore';
const store = configureStore({});
import 'react-redux';


test('рендеринг карточек с информацией', () => {

    const item =  {id: 1, type: "Laptop", img: "https://content2.onliner.by/catalog/device/header/41a120c5ab9e2b54b14c889efb56bb87.jpeg", model: "HP 250 G6 4QW22ES", price: "325$", description: "height-50mm, width-39mm", status: false, cnt: 1};

    const component = renderer.create(
        <Provider store={store}>
            <HashRouter history={hashHistory}>
                <BasketList info={item}/>
            </HashRouter>
        </Provider>

    );

    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();

});