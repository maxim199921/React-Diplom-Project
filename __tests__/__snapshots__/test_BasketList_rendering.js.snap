// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`рендеринг карточек с информацией 1`] = `
<div
  className="col-12 BasketList"
>
  <div
    className="row"
  >
    <div
      className="col-12 col-lg-2 BasketList-img-wrapper"
    >
      <img
        className="BasketList-img"
        src="https://content2.onliner.by/catalog/device/header/41a120c5ab9e2b54b14c889efb56bb87.jpeg"
      />
    </div>
    <div
      className="col-12 col-lg-8 BasketList-info align-self-center"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-12 BasketList-model"
        >
          HP 250 G6 4QW22ES
        </div>
        <div
          className="col-lg-12 BasketList-description"
        >
          height-50mm, width-39mm
        </div>
        <div
          className="BasketList-count"
        >
          <span
            className="BasketList-count-minus"
            onClick={[Function]}
          >
            -
          </span>
          <input
            className="BasketList-count-inp"
            disabled={true}
            type="text"
            value={1}
          />
          <span
            className="BasketList-count-plus"
            onClick={[Function]}
          >
            +
          </span>
        </div>
      </div>
    </div>
    <div
      className="col-6 col-lg-1 BasketList-price align-self-center"
    >
      325
      $
    </div>
    <div
      className="col-6 col-lg-1 align-self-center"
    >
      <input
        className="btn btn-info btn-primary-add"
        onClick={[Function]}
        type="button"
        value="delete"
      />
    </div>
  </div>
</div>
`;
